src/Core/AdminConsole/OrganizationFeatures/Policies/PolicyValidators/AutomaticUserConfirmationPolicyValidator.cs

```diff
@@ -0,0 +1,131 @@
+﻿using Bit.Core.AdminConsole.Entities;
+using Bit.Core.AdminConsole.Enums;
+using Bit.Core.AdminConsole.OrganizationFeatures.Policies.Models;
+using Bit.Core.AdminConsole.OrganizationFeatures.Policies.PolicyUpdateEvents.Interfaces;
+using Bit.Core.AdminConsole.Repositories;
+using Bit.Core.Enums;
+using Bit.Core.Repositories;
+
+namespace Bit.Core.AdminConsole.OrganizationFeatures.Policies.PolicyValidators;
+
+/// <summary>
+/// Represents a validator for the Automatic User Confirmation policy.
+///
+/// This class validates that the following conditions are met:
+/// <ul>
+///     <li>The Single organization policy is enabled</li>
+///     <li>All organization users are compliant with the Single organization policy</li>
+///     <li>No provider users exist</li>
+/// </ul>
+///
+/// This class also performs side effects when the policy is being enabled. They are:
```

Comment:
It looks like the side effect also gets triggered when we disable the policy, so we should update this to reflect that.